drivers/qmc5883l/qmc5883l.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_qmc5883l
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the QMC5883L device driver
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <limits.h>
+
+#include "assert.h"
+#include "periph/gpio.h"
+#include "qmc5883l.h"
+#include "qmc5883l_internal.h"
+
+/* shortcut to the i2c device address */
+#define ADDR        QMC5883L_ADDR
+
+static int _reg_read(const qmc5883l_t *dev, uint8_t reg,
+                     uint8_t *val, int acquire, int release)
```

Comment:
I wonder if we should reuse the concept of flags that are ORed in one parameter instead of using two parameters (I2C does with START and STOP condition). I am not sure what is better for the optimization and if it has any effects at all. I will try it in #10083

Reply:
The effects should be negligible in this case.